tests/api/controllers/gps/gps.c

```diff
@@ -4,59 +4,32 @@
 #include "../../../lib/ts_assertion.h"
 #include "../../../lib/ts_utils.h"
 
-#define TIME_STEP 32
+void check_position(WbDeviceTag gps, double expected_position[], char *end_msg) {
+  const double *position = wb_gps_get_values(gps);
+  char axisNames[] = "XYZ";
+  for (int i = 0; i < 3; i++)
+    ts_assert_double_in_delta(position[i], expected_position[i], 0.000001,
+                              "The %c value measured by the GPS should be %g and not %f %s.", axisNames[i],
```

Comment:
I've noticed usage of %g, why is that ? I've never used it...

Reply:
It's for double with a more human readable format, see https://en.wikipedia.org/wiki/Printf_format_string for details.

Reply:
So we should use the same at the two places, either %g or %f.

Reply:
in tests/api/controllers/gps_speed/gps_speed.c we use %f that for sure

Reply:
I've taken the option of %f